src/main/java/core/basesyntax/WorkWithFile.java

```diff
@@ -1,7 +1,48 @@
 package core.basesyntax;
 
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+
 public class WorkWithFile {
     public void getStatistic(String fromFileName, String toFileName) {
+        int supplySum = 0;
+        int buySum = 0;
+
+        // Чтение данных из файла
+        try (BufferedReader reader = new BufferedReader(new FileReader(fromFileName))) {
+            String line;
+            while ((line = reader.readLine()) != null) {
+                String[] parts = line.split(",");
+                String operationType = parts[0];
+                int amount = Integer.parseInt(parts[1]);
+
+                // Увеличение суммы в зависимости от типа операции
+                if (operationType.equals("supply")) {
+                    supplySum += amount;
+                } else if (operationType.equals("buy")) {
+                    buySum += amount;
+                }
+            }
+        } catch (IOException e) {
+            throw new RuntimeException("Can't write data to file");
+        }
+
+        // Подсчет результата
+        int result = supplySum - buySum;
+
+        // Формирование строки отчета
+        String report = "supply," + supplySum + System.lineSeparator()
+                + "buy," + buySum + System.lineSeparator()
+                + "result," + result + System.lineSeparator();
 
+        // Запись отчета в файл
+        try (BufferedWriter writer = new BufferedWriter(new FileWriter(toFileName))) {
+            writer.write(report);
+        } catch (IOException e) {
+            throw new RuntimeException("Can't save data to file");
+        }
     }
```

Comment:
This method is responsible for both reading from the input file and writing to the output file, as well as processing the data. It would be better to divide this logic into separate methods to improve readability and maintainability. Consider creating methods like `readDataFromFile`, `processData`, and `writeDataToFile`.